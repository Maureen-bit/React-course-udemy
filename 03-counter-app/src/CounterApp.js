import React, { useState } from "react";
import  PropTypes from 'prop-types';

const CounterApp = ( {value = 13} ) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 1 );
        //  Si no tuvieramos acceso al counter podriamos cambiar el estado ejecuntando -> setCounter( ( value ) => value + 1 )
    };

    const handleSubstract = () => {
        setCounter(counter - 1 );
    };

    const handleReset = () => {
        setCounter(value);
    }


    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>
            {/* onClick={handleAdd() => No es correcto poner la referencia de la función con paréntesis dentro del 
            onclick, ya que esto se interpreta en React como ejecución de esta función en el primer renderizado y sin
            paréntesis significa que la función solamente se ejecutará al hacer click en el botón */}
            <button onClick={handleSubstract} >-1</button>
            <button onClick={handleReset} >Reset</button>
            <button onClick={handleAdd} >+1</button>
         </>   
    )
};

CounterApp.propTypes = {
    value: PropTypes.number
}
export default CounterApp;