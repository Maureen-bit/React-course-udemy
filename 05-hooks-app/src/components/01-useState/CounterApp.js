import React, { useState } from 'react';
import './counter.css'

export const CounterApp = () => {
    const [ state , setState] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30,
        counter4: 40
    });

    /* Deestructuro las propiedades del state y extraigo solo las que voy a usar */
    const { counter1, counter2 } = state;
    
  return (
   <>
   <h1>  counter 1 {counter1} </h1>
   <h1>  counter 1 {counter2} </h1>
   <hr />

   <button 
        onClick={() => setState({ 
            /* Como el Ãºnico estado que quiero cambiar es el counter1, utilizo el operador spread(...) para dejar
            los otros estados con su valor actual y que no desaparezcan */
            ...state,
            counter1: counter1 + 1
        })}
        className='btn btn-primary'>
    +1
   </button>
  </>)
};
