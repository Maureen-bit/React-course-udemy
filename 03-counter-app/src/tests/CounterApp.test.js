import React from "react";
import {shallow} from "enzyme";
import '@testing-library/jest-dom'
import CounterApp from "../CounterApp";

describe('Pruebas en componente: CounterApp', () => {
    // se deja este valor por defecto para seguir viedo las opciones que ofrece shallow
    let view = shallow(<CounterApp />);
    beforeEach(() => {
        // Se ejecuta antes de cada prueba
        // se reinicia el componente al estado inicial antes de cada test
        view = shallow(<CounterApp />);
    })

  test('Debe de mostrar el componenten <CounterApp />> correctamente con valores por defecto', () => {
    expect(view).toMatchSnapshot();
  });
  
  test('Debe de mostrar el valor enviado por props al componenten <CounterApp />', () => {
    const value = 2;
    view = shallow(<CounterApp value={value} />);
    expect(view).toMatchSnapshot();
    const valorH2 = view.find('h2').contains(value);
    expect(valorH2).toBeTruthy();
  });

  test('Debe de incrementar con el botón +1', () => {
    // Simular evento click
    view.find('button').at(2).simulate('click');
    const valuePlusOne = view.find('h2').text().trim();
    expect(valuePlusOne).toBe('14');
  });

  test('Debe de decrementar con el botón -1', () => {
    // Simular evento click
    view.find('button').at(0).simulate('click');
    const valuePlusOne = view.find('h2').text().trim();
    /* Como primero se espicho el aumentar uno entonces estaría en 14 y cuando resto volvería 
    al 13, por eso espera el 13, para evitar estos casos usamos el beforeEach para reiniciar el 
    componente
    expect(valuePlusOne).toBe('13');
    */
    expect(valuePlusOne).toBe('12');
  });

  test('Debe de colocar el valor por defecto con el boton reset', () => {
    const value = 90;
    view = shallow(<CounterApp value={value} />);
    view.find('button').at(1).simulate('click');
    const reset = view.find('h2').text().trim();
    expect(reset).toBe('90');
  });
});
