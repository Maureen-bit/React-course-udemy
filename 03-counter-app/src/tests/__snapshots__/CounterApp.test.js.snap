// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente: CounterApp Debe de mostrar el componenten <CounterApp />> correctamente con valores por defecto 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    13
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;

exports[`Pruebas en componente: CounterApp Debe de mostrar el valor enviado por props al componenten <CounterApp /> 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    2
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
